cmd/kubelet/app/server.go

```diff
@@ -253,7 +253,11 @@ func (s *KubeletServer) Run(_ []string) error {
 		glog.Warning(err)
 	}
 
-	client, err := s.createAPIServerClient()
+	var apiclient *client.Client
+	clientConfig, err := s.CreateAPIServerClientConfig()
+	if err == nil {
```

Comment:
I would change this to

if (err != nil) {
  return err
}
apiclient, err = client.New(clientConfig)
...